BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -706,47 +706,46 @@ public override void OnStart(StartState state)
-                    }
-                    else
-                    {
-                        if (((BDExplosivePart)partModules.Current).tntMass > 0)
+                    case "BDExplosivePart":
+                        ((BDExplosivePart)partModule).ParseWarheadType();
+                        if (((BDExplosivePart)partModule).warheadReportingName == "Continuous Rod")
+                            warheadType = WarheadTypes.ContinuousRod;
+                        else
                             warheadType = WarheadTypes.Standard;
+                        break;
+                    case "ClusterBomb":
+                        clusterbomb = ((ClusterBomb)partModule).submunitions.Count;
+                        break;
+                    case "MultiMissileLauncher":
+                        if (weaponClass == WeaponClasses.Bomb)
+                            clusterbomb *= (int)((MultiMissileLauncher)partModule).salvoSize;
                         else
-                            warheadType = WarheadTypes.Kinetic;
-                    }
-                }
-                if (partModules.Current.moduleName == "ClusterBomb")
-                {
-                    clusterbomb = ((ClusterBomb)partModules.Current).submunitions.Count;
-                }
-                if (partModules.Current.moduleName == "MultiMissileLauncher" && weaponClass == WeaponClasses.Bomb)
-                {
-                    clusterbomb *= (int)((MultiMissileLauncher)partModules.Current).salvoSize;
-                }
-                if (partModules.Current.moduleName == "ModuleEMP")
-                {
-                    warheadType = WarheadTypes.EMP;
-                    StandOffDistance = ((ModuleEMP)partModules.Current).proximity;
-                }
-                if (partModules.Current.moduleName == "BDModuleNuke")
-                {
-                    warheadType = WarheadTypes.Nuke;
-                    StandOffDistance = BDAMath.Sqrt(((BDModuleNuke)partModules.Current).yield) * 500;
+                        {
+                            if (warheadType == WarheadTypes.Kinetic) warheadType = WarheadTypes.Launcher;
+                            continue; // Set the warhead type as launcher if not otherwise set, but continue looking for other types.
+                        }
+                        break;
+                    case "ModuleEMP":
+                        warheadType = WarheadTypes.EMP;
+                        StandOffDistance = ((ModuleEMP)partModule).proximity;
+                        break;
+                    case "BDModuleNuke":
+                        warheadType = WarheadTypes.Nuke;
+                        StandOffDistance = BDAMath.Sqrt(((BDModuleNuke)partModule).yield) * 500;
+                        break;
+                    default:
+                        continue;
                 }
-                else continue;
-                break;
+                break; // Break if a valid module is found.
```

Comment:
This stops once a valid module is found, but what should happen if a missile has both `BDExplosivePart` and `ModuleEMP`?

Reply:
EMP detonation is tied to the part the EMP module is on dying, so it'd go off either way, so in that case it should be fine to have it prioritize a warheadtype.Standard/ShapedCharge/ContRod?
Might need to check if the EMP module still arms properly, though.